erpnext/selling/page/point_of_sale/point_of_sale.py

```diff
@@ -17,45 +17,79 @@
 def search_by_term(search_term, warehouse, price_list):
 	result = search_for_serial_or_batch_or_barcode_number(search_term) or {}
 
-	item_code = result.get("item_code") or search_term
-	serial_no = result.get("serial_no") or ""
-	batch_no = result.get("batch_no") or ""
-	barcode = result.get("barcode") or ""
-
-	if result:
-		item_info = frappe.db.get_value(
-			"Item",
-			item_code,
-			[
-				"name as item_code",
-				"item_name",
-				"description",
-				"stock_uom",
-				"image as item_image",
-				"is_stock_item",
-			],
-			as_dict=1,
-		)
+	item_code = result.get("item_code", search_term)
+	serial_no = result.get("serial_no", "")
+	batch_no = result.get("batch_no", "")
+	barcode = result.get("barcode", "")
```

Comment:
Are you aware that 
- `{}.get("barcode", "")` is equal to `{}.get("barcode") or ""`, but
- `{"barcode": None}.get("barcode") or ""` returns `""` while `{"barcode": None}.get("barcode", "")` returns `None`?

So, your changes could produce `None` values, while the former code didn't.

Reply:
Valid point, but `search_for_serial_or_batch_or_barcode_number()` (which use `scan_barcode()`) does not return explicit `None` values.

- https://github.com/frappe/erpnext/blob/develop/erpnext/stock/utils.py#L557
- https://stackoverflow.com/a/6130800/11143333